include/ccf/service/tables/jwt.h

```diff
@@ -37,34 +37,51 @@ namespace ccf
   using JwtIssuer = std::string;
   using JwtKeyId = std::string;
   using Cert = std::vector<uint8_t>;
+  using PublicKey = std::vector<uint8_t>;
 
   struct OpenIDJWKMetadata
   {
-    Cert cert;
+    PublicKey public_key;
     JwtIssuer issuer;
     std::optional<JwtIssuer> constraint;
   };
   DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(OpenIDJWKMetadata);
-  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadata, cert, issuer);
+  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadata, issuer, public_key);
   DECLARE_JSON_OPTIONAL_FIELDS(OpenIDJWKMetadata, constraint);
 
-  using JwtIssuers = ServiceMap<JwtIssuer, JwtIssuerMetadata>;
-  using JwtPublicSigningKeys =
+  using JwtPublicSigningKeysMetadata =
     ServiceMap<JwtKeyId, std::vector<OpenIDJWKMetadata>>;
 
+  struct OpenIDJWKMetadataLegacy
+  {
+    Cert cert;
+    JwtIssuer issuer;
+    std::optional<JwtIssuer> constraint;
+  };
+  DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(OpenIDJWKMetadataLegacy);
+  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadataLegacy, issuer, cert);
+  DECLARE_JSON_OPTIONAL_FIELDS(OpenIDJWKMetadataLegacy, constraint);
+
+  using JwtPublicSigningKeysMetadataLegacy =
+    ServiceMap<JwtKeyId, std::vector<OpenIDJWKMetadataLegacy>>;
+
+  using JwtIssuers = ServiceMap<JwtIssuer, JwtIssuerMetadata>;
+
   namespace Tables
   {
     static constexpr auto JWT_ISSUERS = "public:ccf.gov.jwt.issuers";
 
     static constexpr auto JWT_PUBLIC_SIGNING_KEYS_METADATA =
-      "public:ccf.gov.jwt.public_signing_keys_metadata";
+      "public:ccf.gov.jwt.public_signing_keys_metadata_v2";
 
     namespace Legacy
     {
```

Comment:
Just a general question, I was wondering why the `JWT_PUBLIC_SIGNING_KEYS` and `JWT_PUBLIC_SIGNING_KEY_ISSUER` fields were under `Legacy` tables as they are valid and used for token issuers with manual key refresh. I assume it is a legacy for the automatic key refresh approach and hence was put under legacy but was just curious.

Reply:
Also worth checking #5809 for more details, but in short
* `JWT_PUBLIC_SIGNING_KEYS` and `JWT_PUBLIC_SIGNING_KEY_ISSUER` were used in combination, but were two different tables with key=`kid`, which didn't allow sharing the `kid`s in between the issuer, for example, for Entra multi-tenancy
* They were replaced by `JWT_PUBLIC_SIGNING_KEYS_METADATA`, which was `kid` -> `[(cert, issuer, issuer_constraint), ... ], only this table was written to in the new code, for either manual or auto-refresh, as they both follow the same code path in `jwt_management.h` ([here](https://github.com/microsoft/CCF/blob/main/src/node/rpc/jwt_management.h#L101))
* Now `JWT_PUBLIC_SIGNING_KEYS_METADATA` refers to almost the same structure, but the `cert` got replaced with `public_key`, as long as we now **only** store public keys
* Nit: storing only public key doesn't mean dropping support for certs (x5c), they are kept as a fallback option when we're parsing the JWK documents, but we extract the public key on a fly before saving it into the KV